MapboxDirections/OfflineDirections.swift

```diff
@@ -0,0 +1,174 @@
+import Foundation
+
+
+public typealias OfflineDownloaderCompletionHandler = (_ location: URL?, _ error: Error?) -> Void
+public typealias OfflineDownloaderProgressHandler = (_ bytesWritten: Int64,
+    _ totalBytesWritten: Int64,
+    _ totalBytesExpectedToWrite: Int64) -> Void
+public typealias OfflineVersionsHandler = (_ version: [Version]?, _ error: Error?) -> Void
+
+struct AvailableVersionsResponse: Codable {
+    let availableVersions: [Version]
+}
+
+@objc(MBVersion)
+public class Version: NSObject, Codable {
+    let versionString: String
+    let versionDate: Date
+    
+    static var dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd"
+        return formatter
+    }()
+    
+    convenience init(_ versionDate: Date) {
+        self.init(versionString: Version.dateFormatter.string(from: versionDate), versionDate: versionDate)
+    }
+    
+    convenience init(_ versionString: String) {
+        self.init(versionString: versionString, versionDate: Version.dateFormatter.date(from: versionString)!)
+    }
+    
+    init(versionString: String, versionDate: Date) {
+        self.versionString = versionString
+        self.versionDate = versionDate
+    }
+    
+    required public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        let versionString = try container.decode(String.self)
+        
+        self.versionString = versionString
+        self.versionDate = Version.dateFormatter.date(from: versionString)!
+    }
+    
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.singleValueContainer()
+        try container.encode(versionString)
+    }
+}
+
+@objc(MBOfflineDirections)
+public class OfflineDirections: NSObject, URLSessionDownloadDelegate {
```

Comment:
We could probably fold this into `Directions` if we mention `offline` in the signature and/or documentation.
@1ec5 what’s your take on this?